plugins/org.jboss.reddeer.core/src/org/jboss/reddeer/core/handler/WidgetHandler.java

```diff
@@ -74,6 +74,22 @@ public boolean isEnabled(Widget widget) {
 		}
 		return ret;
 	}
+	
+	/**
+	 * Finds out whether specified {@link Widget} is disposed or not.
+	 * 
+	 * @param widget widget to handle
+	 * @return true if widget is disposed, false otherwise
+	 */
+	public boolean isDisposed(final Widget widget) {
+		return Display.syncExec(new ResultRunnable<Boolean>() {
+
+			@Override
+			public Boolean run() {
```

Comment:
I have no reference to such a problem. For me it's ok to return true when widget is null for me from test point of view if widget is disposed or null it's the same but let's do it your way and don't waste time with any further discussion.